### ISSUE TITLE
MissingJavadocMethod: unexpected violation when scope is private

### ISSUE THREAD

**Issue by Gaurav-Punjabi** [checkstyle/checkstyle#8141]
Check documentation: https://checkstyle.sourceforge.io/config_javadoc.html#MissingJavadocMethod

```
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ javac Test.java 
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ cat config.xml 
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.2//EN" "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="MissingJavadocMethod">
            <property name="scope" value="private"/>
        </module>
    </module>
</module>
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ cat Test.java 
public class Test {
  public void test() {} // should be OK, but violation is thrown
  private void test1() {} // violation (expected)
}
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ java -jar target/checkstyle-8.32-SNAPSHOT-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /Users/gauravpunjabi/Coding/Open Source/checkstyle/Test.java:2:3: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] /Users/gauravpunjabi/Coding/Open Source/checkstyle/Test.java:3:3: Missing a Javadoc comment. [MissingJavadocMethod]
Audit done.
Checkstyle ends with 2 errors.
```

#### Description
When the `scope` of `MissingJavadocMethod` is set to private, it throws a violation for public methods as well. As specified in its documentation, it must only check the fields/methods with the given scope.

#### Expected Output
The check must throw violation only for the given scope. In this example, it must throw only one violation:
```
[ERROR] /Users/gauravpunjabi/Coding/Open Source/checkstyle/Test.java:3:3: Missing a Javadoc comment. [MissingJavadocMethod]
```

**Comment by wltan** [comment]
According to https://checkstyle.sourceforge.io/property_types.html#scope,
> Using a specific scope means not only will that modifier be examined, but also all the modifiers listed above it in the previous list.

So a scope setting of `private` will check for `public`, `protected`, and `package` methods as well.

**Comment by Gaurav-Punjabi** [comment]
Got it, thanks 👍 I'm closing the issue
